### PR TITLE
Added validation of IPv6 addresses and supporting tests

### PR DESCRIPTION
**Pull request by tomeon**
Did a pretty extensive restructuring of the code, too -- may have gone a bit overboard, but all tests are passing so at least there's that :).

The IPv6 validation is not foolproof -- it doesn't check for things like non-hex characters, spaces, etc.  It just enforces RFC 5952's rules about:
1. Compressing only the largest run of all-zero segments;
2. Compressing only the leftmost of the all-zero runs if there are more than one runs of equal length; and
3. Not compressing a single zero segment.

The validator routine also checks that the address contains at most eight segments.

Would love to hear feedback, and would be more than happy to make any suggested changes.  Thanks in advance!

### CODE REVIEW

**Comment by jonathanio** [comment]
This is fantastic. I wasn't expecting anything like this, this fast, and it's brilliant to see the test framework being put to good use too! I'll take a look over the next few days and see if there's anything I can input. Failing that I'll merge and release :+1: 

Thank you for your hard work.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+PREFIX := /etc/openvpn
+DESTDIR := $(PREFIX)/scripts
+
+SRC = update-systemd-resolved
+DEST = $(DESTDIR)/$(SRC)
+
+.PHONY: all install info
+
+all: install info
```
Comment (jonathanio) [inline Makefile:9]: Out of interest, is there any reason you didn't include a `test` stage in the Makefile?

**Comment by jonathanio** [comment]
@BaxterStockman I've taken the time to look through the changes, and they're great. Just the one small question which I've noted above. I'll await your feedback on that before merging and releasing.

File: Makefile (hunk 1 above)
Comment (tomeon) [inline Makefile:9]: Whoops, nope -- just didn't think of it.

**Comment by tomeon** [comment]
@jonathanio -- sorry for taking so long to get back to you.  I have updated the `Makefile` with a `test` target, and also added a check for compliance with RFC 5952 section 4.2.1.

Re: the testing framework -- it was nice to work with something much faster and less confounding than [`bats`](https://github.com/sstephenson/bats), the shell testing framework I hate to love :).

One other small thing -- I changed the Makefile to suggest configuring the script to run during the `pre-down` phase rather than the `down` phase -- I saw errors in the journal when using the `down` phase because the VPN interface was already down by the time `systemd-resolved` came along to update the DNS servers.

**Comment by jonathanio** [comment]
@BaxterStockman not a problem. I'll get this all merged and released. Thank you so much for your work on this.

**Pull request merged**